services/src/main/java/org/keycloak/protocol/oidc/endpoints/IframeUtil.java

```diff
@@ -38,7 +44,7 @@ public static Response returnIframeFromResources(String fileName, String version
             cacheControl = CacheControlUtil.noCache();
         }
 
-        InputStream resource = IframeUtil.class.getResourceAsStream(fileName);
+        Object resource = responseEntityProvider.get();
```

Comment:
Is it normal for this to be typed as `Object`? Or is there a more specific type available?

Reply:
Yes, you can see that method `jakarta.ws.rs.Resource.ok` takes `Object` as an argument. You can pass various types of objects here like the:
- String (That is what is used in the case of freemarker template)
- InputStream (That is what is used in the case of `IframeUtil.returnIframeFromResources` which is used for loading html files https://github.com/keycloak/keycloak/pull/34551/files#diff-1bad86b3a44fb4631eddecd1d9f7ec38f93a6450a8161ad2babe8163ef59b2afR33 )
- Some more types (You can see Keycloak sources for the inspiration of what we're passing here in various places)

Reply:
Thanks for the clarification!